contribs/drt/src/main/java/org/matsim/contrib/drt/optimizer/rebalancing/Feedforward/FeedforwardRebalancingStrategy.java

```diff
@@ -124,8 +125,10 @@ public FeedforwardRebalancingStrategy(DrtZonalSystem zonalSystem, Fleet fleet, R
 				// vehileToSend after scaling.
 				int numAvailableVehiclesInZone = 0;
 				if (rebalancableVehiclesPerZone.get(departureZone) != null) {
-					numAvailableVehiclesInZone = (int) rebalancableVehiclesPerZone.get(departureZone).stream()
-							.filter(v -> !relocatedVehicles.contains(v)).count();
+					List<DvrpVehicle> actualRebalancableVehicles = rebalancableVehiclesPerZone.get(departureZone)
+							.stream().filter(v -> !relocatedVehicles.contains(v)).collect(Collectors.toList());
```

Comment:
The map with lists of rebalancable vehicles should be initialised only once, i.e. before the `for` loop starts.

Reply:
In a broader context (beyond the lines modified in this PR):
- call `rebalancableVehiclesPerZone.get(departureZone)` only once inside the loop and store in a local variable
- removing vehicles from the beginning of an array list is a bit costly, consider removing the last one from the list (line 141-142)